fetch/fetch.go

```diff
@@ -452,11 +452,10 @@ func (f *Fetch) organizeRequests(requests []requestMessage) map[p2p.Peer][][]req
 	peer2requests := make(map[p2p.Peer][]requestMessage)
 
 	for _, req := range requests {
-		hashPeersMap, exists := f.hashToPeers.Get(req.Hash)
+		hashPeers, exists := f.hashToPeers.GetList(req.Hash)
```

Comment:
btw, AddPeersFromHash() suffer the same problem, concurrent map iteration and map write.

Reply:
made mutex public + now using it in `AddPeersFromHash`